dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorArm8.java

```diff
@@ -141,6 +141,7 @@ public List<Event> visitAtomicFetchOp(AtomicFetchOp e) {
         Store store = newRMWStoreExclusive(address, dummyReg, Tag.ARMv8.extractStoreMoFromCMo(mo), true);
         Label label = newLabel("FakeDep");
         Event fakeCtrlDep = newFakeCtrlDep(resultRegister, label);
+        fakeCtrlDep.addFilters(Tag.NOOPT);
```

Comment:
Why not integrate the addition of the tag directly into the function `newFakeCtrlDep`?